wordpress-mobile/gutenberg-mobile
# Fix scroll position on iOS

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/197 partially

This change fixes inconsistent scroll behavior with input texts on iOS. Android behavior didn't change since it wasn't problematic.

iOS Before:
![ezgif com-resize-2](https://user-images.githubusercontent.com/5032900/50084526-5a549f80-0208-11e9-83ad-030ee445b9e8.gif)

After:

![ezgif com-video-to-gif-2](https://user-images.githubusercontent.com/5032900/50087919-98ef5780-0212-11e9-89f3-83cfe51684c8.gif)

**Testing prerequisites**

For WPiOS

- Checkout the PRs branch to any arbitrary folder and cd .. to it
- run `yarn install`, `yarn start:reset`
- Open XCode WPiOS on the latest develop
- Clean build folder on Xcode, and then run the app

For Android

- open grade.properties at WordPress-Android folder
- add `wp.BUILD_GUTENBERG_FROM_SOURCE = true` to grade.properties
- checkout the PRs branch in the subrepo of WordPress-Android repo
- cd to WordPress-Android/libs/gutenberg-mobile
- run `yarn install`, `yarn start:reset`
- `yarn wpandroid` on a separate terminal in the same directory

**To Test**

Test 1 - Empty post on WPiOS with return button

- Open an empty post via WPiOS app
- Start typing and press return button to create a new block
- Repeat this until you add about 10 paragraph blocks
- Verify that typed text input is viewable
- Close the keyboard
- Verify that scroll position goes back to normal

Test 2 - Add block with (+) button
- Open some profiled post via WPiOS app
- Add new text blocks with (+) button
- Verify that recently added text input is viewable

Test 3 - Verify inner block buttons are working

- Use up&down&trash buttons when keyboard is open or closed
- Verify that they are working

Test 4 - Repeat tests for Android and make sure we didn't break anything, Android behavior shouldn't change with this PR.
